### ISSUE TITLE
Fixar tabelas do banco

### ISSUE THREAD

**Issue by leomp12** [ecomplus/app-tiny#7]
Alguns erros com as tabelas atuais neste arquivo `db-dump.sql`:

```console
ERROR 1067 (42000): Invalid default value for 'created_at'
```

Deveria set TIMESTAMP em vez de DATETIME, talvez seja necessário alterar no source code onde você trata esse campo.

```console
ERROR 1293 (HY000): Incorrect table definition; there can be only one TIMESTAMP column with CURRENT_TIMESTAMP in DEFAULT or ON UPDATE clause
```

> Só pode haver uma tabela com `CURRENT_TIMESTAMP`, realmente precisa de `created_at` e `updated_at` aqui? Só um não resolveria?

```sql
`updated_at` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
```

Não cheguei a testar as outras tabelas, mas vai acontecer os mesmos erros (pelo menos).

Além disto, está usando INT desnecessariamente em todos os campos em vez das variações (TINYINT, MEDIUMINT) conforme a necessidade do campo.

Por último, podemos limpar este arquivo de dump automático pra termos apenas um arquivo SQL com o que realmente é necessário, apenas criando as tabelas já com os modelos otimizados e setando o valor do auto increment logo na criação da respectiva tabela.

**Comment by leomp12** [comment]
@talissonf pode dar uma olhada nisso por favor?

**Comment by leomp12** [comment]
Também considere adicionar INDEX(s) se for válido, provavelmente seria útil para o `_id`, campo(s) muito utilizados como filtro de busca.

Talvez ajude como referência:
https://github.com/ecomclub/procedures/blob/master/assets/tables.sql

**Comment by talissonf** [comment]
Esse dump tá com essa tipagem porque o db que o heroku disponibiliza não tinha vários tipos, por isso que te falei ontem que esse dump aí não tava utilizável!

**Comment by leomp12** [comment]
@talissonf abri o issue antes da sua resposta sobre isto ontem.

De qualquer forma acho que isto não deve ser feito com dump automático.

E talvez estas correções das tabelas causem problemas no source do app, então abri o issue pra você ver primeiro.

**Comment by leomp12** [comment]
@talissonf ainda temos problemas aqui:

1. Continua sendo um dump com queries não desejadas (não é pra setar o time zone do MySQL aqui por exemplo);
2. Ainda tem `ALTER TABLE`s desnecessárias, você pode setar o auto increment no `CREATE TABLE`;
3. Do jeito que está o `updated_at` vai ficar sempre NULL, não precisa dele? Em nenhuma ocasião você precisa saber o momento em que a row foi atualizada? Se realmente não precisar, o campo nem deveria existir.

**Comment by talissonf** [comment]
aqui ? https://github.com/ecomclub/app-tiny/blob/master/assets/db-dump.sql

**Comment by leomp12** [comment]
Sim, dei uma olhada nos commits e não vi criando as tabelas em outros lugar, criou?

**Comment by talissonf** [comment]
Eu criei na mão mesmo pra testar o app. To mexendo com outra coisa e não queria parar o que eu to fazendo pra mexer com o tiny. Ta funcionando, depois altero esse dump e deixo as tabelas pra serem criadas ou alteradas pelo app.

**Comment by leomp12** [comment]
Mas você criou na mão pra mockar no seu local, certo?
O app não funciona enquanto estas tabelas não estiverem criadas no servidor, então não temos como testá-lo em "live"...

**Comment by talissonf** [comment]
as tabelas foram criadas no servidor, leo!

**Comment by leomp12** [comment]
Legal, então quando o código estiver correto, fechamos o issue.

**Comment by talissonf** [comment]
Ok.
